src/input_arguments_handling.jl

```diff
@@ -0,0 +1,122 @@
+        closequotechar=UInt8(closequotechar),
+        escapechar=UInt8(escapechar),
+        delim=delim,
+        quoted=quoted,
+        stripwhitespace=stripwhitespace,
+        trues=truestrings,
+        falses=falsestrings,
+        groupmark=groupmark,
+        dateformat=dateformat,
+        ignorerepeated=ignorerepeated,
+        decimal=UInt8(decimal),
+        ignoreemptylines=ignoreemptyrows,
+        rounding=rounding,
+    )
+end
+
+# Check that the schema and header are compatible
+_validate(header::Vector, schema::Vector, validate_type_map) = length(header) == length(schema) || throw(ArgumentError("Provided header and schema lengths don't match. Header has $(length(header)) columns, schema has $(length(schema))."))
+_validate(header::Vector, schema::Dict{Symbol}, validate_type_map) = !validate_type_map || issubset(keys(schema), header) || throw(ArgumentError("Provided header and schema names don't match. In schema, not in header: $(collect(setdiff(keys(schema), header))). In header, not in schema: $(setdiff(header, keys(schema)))"))
+function _validate(header::Vector, schema::Dict{Int}, validate_type_map)
+    validate_type_map || return
+    len = length(header)
+    (lo, hi) = extrema(keys(schema))
+    (lo < 1 || hi > len) && throw(ArgumentError("Provided schema indices are incompatible with header of length $(len). Offending indices from schema $(collect(setdiff(keys(schema), 1:len))))."))
+end
+_validate(header, schema, validate_type_map) = true
+
+_nbytes(::UInt8) = 1
+_nbytes(x::Union{String,Char}) = ncodeunits(x)
+
+# We're a bit stricter than Parsers.jl here, because the Lexer doesn't support
+# multi-byte characters.
+function validate_parser_args(;openquotechar, closequotechar, delim, escapechar, decimal, newlinechar, ignorerepeated)
+    _nbytes(openquotechar) == 1 || throw(ArgumentError("`openquotechar` must be a single-byte character"))
+    _nbytes(closequotechar) == 1 || throw(ArgumentError("`closequotechar` must be a single-byte character"))
+    if isnothing(delim)
+        ignorerepeated && throw(ArgumentError("auto-delimiter detection not supported when `ignorerepeated=true`; please provide delimiter like `delim=','`"))
+    else
+        _nbytes(delim) == 1 || throw(ArgumentError("`delim` must be a single-byte character"))
+    end
+    _nbytes(escapechar) == 1 || throw(ArgumentError("`escapechar` must be a single-byte character"))
+    _nbytes(decimal) == 1 || throw(ArgumentError("`decimal` must be a single-byte character"))
+
+    if !isnothing(newlinechar)
+        _nbytes(newlinechar) > 1 && throw(ArgumentError("`newlinechar` must be a single-byte character."))
+        ((newlinechar % UInt8) in ((openquotechar % UInt8), (closequotechar % UInt8), (escapechar % UInt8), (something(delim, 0x00) % UInt8))) &&
+            throw(ArgumentError("`newlinechar` must be different from `delim`, `openquotechar`, `closequotechar`, and `escapechar`"))
+    end
+
+    return nothing
+end
```

Comment:
do we have test cases for everyone of these error conditions? would be good to double-check that (and open an issue to add any that are missing)

Reply:
Yeah there are tests for this in `test/exception_handling.jl`